invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -310,6 +310,8 @@ def get_form_config(**kwargs):
             ]
         ),
         user_communities_memberships=get_user_communities_memberships(),
+        # TODO: Name should be snake-case, i.e. "can_have_metadata_only_records"
+        canHaveMetadataOnlyRecords=conf.get("RDM_RECORDS_ALLOW_METADATA_ONLY"),
```

Comment:
- I think for other config-based components we parse snake_case, and assign to a camelCase prop 🤷 
- We've had discussions in the past that we're "polluting" this config with all sorts of variables. Ideally, we should use Overridable where possible. 
  - In this case specifically, this is not only a UI/styling change though, it actually comes/applies from the backend (similar to the PIDs config, and vocabularies)